### ISSUE TITLE
panel buggy on multiple xscreens

### ISSUE THREAD

**Issue by ethanj** [mate-desktop/mate-panel#149]
I just did a clean install of mint 16, and configured a second xscreen by modifying xorg.conf .
When mate comes up, it correctly has a panel at the bottom of each screen, and I can add widgets to each.
However, any interaction with the widgets on the second screen shows up on the first screen rather than the second.
For instance, if I add a menu to the second screen's panel, then click on it, the popup which would normally show up above it shows up in the exact same location on the first screen, even if the first screen's panel does not contain a menu widget.
##

**Comment by monsta** [comment]
@ethanj: this should be already fixed in 1.10 (see https://github.com/mate-desktop/mate-panel/commit/3f67cbdfeaa53820ed690dd25f52dc162e7511e2).

**Comment by axelolmos** [comment]
I still had the problem with 1.10 (mate-panel-1.10.1-1.fc22.x86_64) and had to apply the patch I wrote.

**Comment by monsta** [comment]
Well that's because your issue is different.

**Comment by axelolmos** [comment]
This issue has gotten worse since Fedora 20.  In Fedora 20, multiple DISPLAYs would work with the mate-panel patch I proposed.  After that, (Fedora 21,22,23...) my patch fixes the menu, but on DISPLAYS greater than :0:0 the dropdown menu doesn't actually work, and the screen has a black background.

It really seems like no testing of DISPLAY > :0.0 is being done, and as new code is introduced, multiple DISPLAYs are breaking more and more.  I know testing multiple monitors, and drivers other than nouveau (e.g. NVIDIA) is a pain, but other modern OS's (Windows, Mac) support multiple monitors fine.

**Comment by raveit65** [comment]
> the dropdown menu doesn't actually work
You mean hovering with the mouse over a menu entry does not work in f23?

**Comment by axelolmos** [comment]
Yes, the menu pops down, but doesn't actually work.  In addition, the redraw of the menus does not work, so they just make a mess after they drop down as the desktop background is black and does not get updated.

I tested most of the desktops for Fedora 23:
NVIDIA Driver, Separate X Screens in xorg.conf:

KDE Plasma Workspaces           Black screen on 2nd screen
LXDE Desktop                            WORKS!  Needs menu bar on all windows.  
LXQt Desktop                             WORKS!  Needs menu bar on all windows.
Cinnamon Desktop                     Fallback mode, Grey screen on 2nd screen
MATE Desktop                           Black screen on 2nd screen, menu appears but doesn't work
Basic Desktop                            Black screen on 2nd screen.
Xfce Desktop                             WORKS!  Needs menu bar on all windows.

After some testing and tweaking I was able to get Xfce working in a manner similar to MATE.
It has the menu bar / panel on all monitors.  I disabled the middle mouse button which was popping up the menu on the wrong $DISPLAY.

It seems the Gnome forked variants are heading down the road of not supporting separate X screens, which unfortunately, I require.  As such, I am switching to Xfce as it was the closest to MATE, but supports multiple X screens.

**Comment by raveit65** [comment]
Does this issue still exists?
